wandb/sdk/data_types/image.py

```diff
@@ -148,12 +149,23 @@ def __init__(
         if isinstance(data_or_path, Image):
             self._initialize_from_wbimage(data_or_path)
         elif isinstance(data_or_path, str):
-            self._initialize_from_path(data_or_path)
+            if Image.path_is_reference(data_or_path):
+                self._path = data_or_path
+                self._is_tmp = False
+                self._sha256 = hashlib.sha256(data_or_path.encode("utf-8")).hexdigest()
+            else:
+                self._initialize_from_path(data_or_path)
+            ext = os.path.splitext(data_or_path)[1][1:]
+            self.format = ext
```

Comment:
`os.path.splitext` conceptually takes a filepath, not a URL, and there might be strange cases where the parsing differs; for example, StackOverflow avatars have URLs like `https://i.stack.imgur.com/hDsZP.jpg?s=64&g=1`, and the query string would get included in the `splitext` extension. So I think `os.path.splitext` should stay in `_initialize_from_path`; if `data_or_path` is a reference, I think it should be something like

```
path = urllib.parse.urlparse(data_or_path).path
ext = path.split("/")[-1].split(".")[-1]  # NOT os.path.splitext, because we always want to split on `/`, even on Windows
self.format = ext
```